src/common/vector/auxiliary_buffer.cpp

```diff
@@ -0,0 +1,53 @@
+#include "common/vector/auxiliary_buffer.h"
+
+#include "common/in_mem_overflow_buffer_utils.h"
+#include "common/vector/value_vector.h"
+
+namespace kuzu {
+namespace common {
+
+void StringAuxiliaryBuffer::addString(
+    common::ValueVector* vector, uint32_t pos, char* value, uint64_t len) const {
+    assert(vector->dataType.typeID == STRING);
+    auto& entry = ((ku_string_t*)vector->getData())[pos];
+    InMemOverflowBufferUtils::copyString(value, len, entry, *inMemOverflowBuffer);
+}
+
+ListAuxiliaryBuffer::ListAuxiliaryBuffer(
```

Comment:
It will cause an dependency issue if i moved it to the header.
We don't include valueVector in the header file, so we are not able to do make_unique<ValueVector> in the header.